module/Finna/src/Finna/RecordDriver/SolrLrmi.php

```diff
@@ -331,20 +341,42 @@ public function getAllImages($language = 'fi', $includePdf = true)
 
         // Attempt to find a PDF file to be converted to a coverimage
         if ($includePdf && empty($result) && $materials = $this->getMaterials()) {
-            foreach ($materials as $material) {
-                if ($material['format'] === 'pdf') {
-                    $url = $material['url'];
-                    $result[] = [
-                        'urls' => [
-                            'small' => $url,
-                            'medium' => $url,
-                            'large' => $url
-                        ],
-                        'description' => '',
-                        'rights' => []
-                    ];
-                    break;
+            // First, find PDF-materials
+            $pdfs = array_filter(
+                $materials,
+                function ($material) {
+                    return $material['format'] === 'pdf';
                 }
+            );
+
+            // ... if not found, try materials that have been converted to PDF
+            if (empty($pdfs)) {
+                $pdfs = array_filter(
+                    $materials,
+                    function ($material) {
+                        return !empty($material['pdfUrl'])
+                            && in_array(
+                                $material['format'],
+                                $this->previewableConvertedFileFormats
+                            );
+                    }
+                );
+            }
+            $pdfs = array_values($pdfs);
+
+            if (!empty($pdfs)) {
+                $pdf = $pdfs[0];
+                // Prefer original PDF-files over converted ones
```

Comment:
Kommentti on vähän hämmentävä? Jos ymmärsin oikein, tässä ei kuitenkaan valita kahden PDF-version väliltä, vaan poimitaan URL oikeasta paikasta?

Reply:
Ehkä "Prefer original PDF file to a converted version"

Reply:
Tai over 🙂